test/unity/toolkits/drawing_classifier/test_dc_serialization.cxx

```diff
@@ -176,8 +185,211 @@ BOOST_AUTO_TEST_CASE(test_export_coreml) {
                    target + "Probability");
 }
 
+BOOST_AUTO_TEST_CASE(test_load_version) {
+  turi::drawing_classifier::drawing_classifier dc;
+  TS_ASSERT_EQUALS(dc.get_version(), dc.DRAWING_CLASSIFIER_VERSION);
+}
+
 BOOST_AUTO_TEST_CASE(test_save_load) {
-  // TODO
+  auto remove_if_exist = [](const char* fname) {
+    std::FILE* fp = nullptr;
+    if ((fp = std::fopen(fname, "r")) != NULL) {
+      if (std::fclose(fp) != 0) {
+        std::stringstream ss;
+        ss << "cannot close file: " << fname << std::endl;
+        throw(turi::error::io_error(ss.str()));
+      }
+
+      if (std::remove(fname) != 0) {
+        std::stringstream ss;
+        ss << "cannot remove file: " << fname << std::endl;
+        throw(turi::error::io_error(ss.str()));
+      }
+    }
+  };
+
+  /**
+   * The test calls `save_impl` first and write serialized initial
+   * `nn_spec_` to disk, let's say `file_1`.
+   *
+   * Then we load the previously stored spec file and update
+   * all params to the model spec. Then we store what's loaded
+   * to `file_2`.
+   *
+   * By comparing `file_1` and `file_2`, we can verify that the
+   * both model spec instances have the same content and same content
+   * is written to the disk during the serialization.
```

Comment:
This only tests that the two saved files are the same, but it doesn't test that `save` works properly. For example, implementing `save` in a way that writes nothing at all would yield equivalent (incorrect) archives.

Reply:
Yeah, that's why we also check the nn_spec in memory as well.